core/src/utils/overlays.ts

```diff
@@ -743,6 +743,37 @@ export const createDelegateController = (ref: {
   };
 };
 
+/**
+ * Creates a lock controller.
+ */
+export const createLockController = () => {
```

Comment:
Can we rename this to `src/utils/lock-controller.ts` and then create a tech debt ticket to update `ion-router-outlet` to use this? (be sure to add a TODO comment inside of the router outlet implementation).

The code is the same, so would be an easy win for us internally.

Reply:
Also you can add "needs_refinement" to the ticket so we take care of it in an upcoming sprint